src/render/tutorials/api-gateway-security-oauth2/2_0.html.md

```diff
@@ -0,0 +1,469 @@
+        .setQueryParam("client_secret", clientSecret)
+        .setQueryParam("redirect_uri", redirectUri)
+        .setQueryParam("grant_type", "authorization_code")
+        .putHeader("Content-Length", "0");
+  }
+
+  private Single<HttpResponse<Buffer>> fetchUserInfo(String accessToken) {
+    return webClient.getAbs(USERINFO_URL)
+        .bearerTokenAuthentication(accessToken)
+        .rxSend();
+  }
+
+}
+
+```
+
+Let's take a look at this file step by step:
+
+1. The constructor is quite straightforward, we create a new [`webClient`](https://vertx.io/docs/vertx-web-client/java/) and save the config for later use.
+2. When a request hits our handler, the `handle` method is invoked. We try to obtain the authorization code from the request.
+3. Then we try to `exchangeCodeForToken`. The first step is to `prepareTokenRequest`. This time we have to provide the `code`, `client_id`, `client_secret`, `redirect_uri`, `grant_type` parameters and `Content-Length` header (this will be a `POST` request). For more details on those parameters see [this documentation](https://developers.google.com/identity/protocols/OAuth2WebServer#exchange-authorization-code). We send the request using `rxSend` method (for more info about Vert.x's RxJava 2 API check [this documentation](https://vertx.io/docs/vertx-web-client/java/#_rxjava_2_api)) and map the response to the acquired access token.
+4. We map the resulting observable with the token to `fetchUserInfo` which creates a `GET` request to Google API with the bearer token authentication and sends it.
+5. Finally we subscribe for the whole processing chain and end our response with a personalized *hello* message!
+
+As always, let's not forget to register our factories in the `resources/META_INF/services` directory!
+
+`io.knotx.server.api.handler.RoutingHandlerFactory`
+```
+io.knotx.examples.security.routing.GoogleLoginRedirectHandlerFactory
+io.knotx.examples.security.routing.OAuth2HelloHandlerFactory
+```
+
+## Build & Run
+
+Now let's see how it works.
+
+In this tutorial we'll be using the Docker distribution, but it will work with Stack distribution as well. Please refer to [this tutorial](http://knotx.io/tutorials/getting-started-with-knotx-stack/2_0/) on how to work with the Stack distribution.
+
+First, let's rename the Docker image we're about to create:
+
+`gradle.properties`
+```
+// ...
+docker.image.name=knotx-example/secure-api-gateway
+// ...
+```
+
+Now let's build the image:
+
+```
+$ gradlew clean build-docker
+```
+
+and run it:
+
+```
+$ docker run -p8092:8092 knotx-example/secure-api-gateway
+```
+
+After a while the Docker container should be up and running.
+
+Now let's type in `http://localhost:8092/api/secure/login` in your browser.
+
+![Select Google Account](/img/blog/api-gateway-security/security-google-select-account.png)
+
+Select your account.
+
+![Select Google Account](/img/blog/api-gateway-security/security-google-hello-message.png)
```

Comment:
Are there any shared assets between those 3 blogposts?
If not, maybe they should be placed in proper folders for each post?